src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -409,6 +409,11 @@ public override BoundNode VisitObjectInitializerMember(BoundObjectInitializerMem
                 Error(ErrorCode.ERR_DictionaryInitializerInExpressionTree, node);
             }
 
+            if (node.MemberSymbol is PropertySymbol property)
```

Comment:
Is there a reason that this doesn't go through `VisitPropertyAccess`? I'm curious if there are other places in the compiler where `VisitPropertyAccess` could be missing a property in this case similar to what was missed here. I did a quick audit of all the `VisitPropertyAccess` and checked to make sure that the corresponding logic also existed in `VisitObjectInitializer`. Seems like we're okay but worries me a bit we could have future holes here.

Reply:
> Is there a reason that this doesn't go through VisitPropertyAccess? 

In this scenario there is no BoundPropertyAccess in the tree.

---
In reply to: [533889046](https://github.com/dotnet/roslyn/pull/49719#discussion_r533889046) [](ancestors = 533889046)